pkg/util/strategicpatch/patch.go

```diff
@@ -66,8 +69,8 @@ type errConflict struct {
 	message string
 }
 
-func newErrConflict(patch, current []byte) errConflict {
-	s := fmt.Sprintf("patch:\n%s\nconflicts with current:\n%s\n", patch, current)
+func newErrConflict(patch, current string) errConflict {
+	s := fmt.Sprintf("patch:\n%s\nconflicts with automated changes:\n%s\n", patch, current)
```

Comment:
Re: the assumption that it was an automated process, we require that clients know about and use the annotation, automated processes do not. No magic there. It's part of the interface, and documented as such.

Re: version skew, good question. Wrote up #16543 and then did some investigation. See the results there. Bottom line, we need to a version check that returns an error if the modified version is older than the original version.